src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1918,8 +1918,22 @@ RouterJob(Query *originalQuery, PlannerRestrictionContext *plannerRestrictionCon
 		 */
 		if (updateOrDeleteOrMergeRTE->rtekind == RTE_SUBQUERY)
 		{
```

Comment:
I'm trying to understand what does it mean for `ExtractResultRelationRTE` to be a subquery.

I guess it means the query is zero shards? Could use some comment on why we have such a weird convention :)

Reply:
For a RTE, if all of the shards are pruned, we replace the relation RTE into subquery RTE that returns no results, using `PlanRouterQuery()->UpdateRelationToShardNames()->ConvertRteToSubqueryWithEmptyResult()`.